pkg/api/routes.go

```diff
@@ -47,11 +48,12 @@ const (
 )
 
 type RouteHandler struct {
-	c *Controller
+	c              *Controller
+	demandedImages sync.Map
```

Comment:
Thinking about it, not sure weather sync.Map is overkill. Maybe a normal map with an internal mutex would suffice.